dowhy/causal_estimators/causalml.py

```diff
@@ -0,0 +1,108 @@
+        self.identifier_method = self._target_estimand.identifier_method
+        self.logger.debug("The identifier method used {}".format(self.identifier_method))
+
+        # Check the backdoor variables being used
+        self.logger.debug("Back-door variables used:" +
+                          ",".join(self._target_estimand.backdoor_variables))
+        
+        # Add the observed confounders and one hot encode the categorical variables
+        self._observed_common_causes_names = self._target_estimand.backdoor_variables
+        if self._observed_common_causes_names:
+            # Get the data of the unobserved confounders
+            self._observed_common_causes = self._data[self._observed_common_causes_names]
+            # One hot encode the data if they are cateogorical
+            self._observed_common_causes = pd.get_dummies(self._observed_common_causes, drop_first=True)
+        else:
+            self._observed_common_causes = []
+        
+        # Check the instrumental variables involved
+        self.logger.debug("Instrumental variables used:"+
+                        ",".join(self._target_estimand.instrumental_variables))
+        
+        # Perform the same actions as the above
+        self._instrumental_variable_names = self._target_estimand.instrumental_variables
+        if self._instrumental_variable_names:
+            self._instrumental_variables = self._data[self._instrumental_variable_names]
+            self._instrumental_variables = pd.get_dummies(self._instrumental_variables, drop_first=True)
+        else:
+            self._instrumental_variables = []
+
+        #Check if effect modifiers are used
+        self.logger.debug("Effect Modifiers used:" + 
+                        ",".join(self._effect_modifier_names))
+        
+        
+        # Get the class corresponding the the estimator to be used
+        estimator_class = self._get_causalml_class_object(self._causalml_methodname)
+        # Initialize the object
+        # pdb.set_trace()
+        self.estimator = estimator_class(**self.method_params["init_params"])
+        self.logger.info("INFO: Using CausalML Estimator")
+        self.symbolic_estimator = self.construct_symbolic_estimator(self._target_estimand)
+        self.logger.info(self.symbolic_estimator)
+        
+    def _get_causalml_class_object(self, module_method_name, *args, **kwargs):
+        
+        try:
+            (module_name, _, class_name) = module_method_name.rpartition(".")
+            estimator_module = import_module(module_name)
+            estimator_class = getattr(estimator_module, class_name)
+
+        except (AttributeError, AssertionError, ImportError):
+            raise ImportError('Error loading {}.{}. Double-check the method name and ensure that all econml dependencies are installed.'.format(module_name, class_name))
+        return estimator_class
+
+    def _estimate_effect(self):
+        X_names = self._instrumental_variable_names + \

```

Comment:
This is a subtle point. We should not pass the instrumental variables to causalml because it simply conditions on them. Conditioning on IVs creates bias. The IV causes the treatment, and treatment causes outcome. If we keep the IV constant, we risk losing all variation in the treatment, and then the outcome may also appear constant.

So I recommend removing iv_names from this line 73.

Reply:
It makes sense. However, it gives us a chance to compare the performance of an informed decision, vs blindly applying estimators.

Reply:
Yes, that's good for comparing estimators, but not so good for a user of dowhy. We should try to get the best effect estimate possible for them. And if the user has told us that there are some variables that are IVs, we should respect that and not pass them to the causalml estimator.
Also, for testing, you can specify (incorrectly) all variables as "common_causes" in the Model() init function and that should be enough to test the scenario you mention.

Reply:
Makes sense. I'll make the change to the source code.